lib/s3middleware/objectLegalHold.js

```diff
@@ -0,0 +1,111 @@
+const { parseString } = require('xml2js');
+const errors = require('../errors');
+
+/*
+    Format of the xml request:
+    <LegalHold>
+        <Status>ON|OFF</Status>
+    </LegalHold>
+*/
+
+/**
+ * validate status - validates status xml
+ * @param {object} status - parsed status object
+ * @return {boolean} - true or false on success, error on failure
+ */
+function _validateStatus(status) {
+    const validatedStatus = {};
+    const expectedValues = ['OFF', 'ON'];
+    if (!status || status[0] === '') {
+        validatedStatus.error = errors.MalformedXML.customizeDescription(
+            'request xml does not contain Status');
+        return validatedStatus;
+    }
+    if (status.length > 1) {
+        validatedStatus.error = errors.MalformedXML.customizeDescription(
+            'request xml contains more than one Status');
+        return validatedStatus;
+    }
+    if (!expectedValues.includes(status[0])) {
+        validatedStatus.error = errors.MalformedXML.customizeDescription(
+            'Status request xml must be one of "ON", "OFF"');
+        return validatedStatus;
+    }
+    validatedStatus.status = status[0];
+    return validatedStatus.status === 'ON';
+}
+
+/**
+ * validate legal hold - validates legal hold xml
+ * @param {object} parsedXml - parsed legal hold xml object
+ * @return {boolean} - contains legal hold status on success,
+ * error on failure
+ */
+function _validateLegalHold(parsedXml) {
+    const validatedLegalHold = {};
+    if (!parsedXml) {
+        validatedLegalHold.error = errors.MalformedXML.customizeDescription(
+            'request xml is undefined or empty');
+        return validatedLegalHold;
+    }
+    if (!parsedXml.LegalHold) {
+        validatedLegalHold.error = errors.MalformedXML.customizeDescription(
+            'request xml does not contain LegalHold');
+        return validatedLegalHold;
+    }
+    const validatedStatus = _validateStatus(parsedXml.LegalHold.Status);
+    if (validatedStatus.error) {
```

Comment:
Does this not error if validatedStatus returned a bool? Like bool has no property? Or JS is just like "meh whatever"?

Reply:
As far as I tested it JS was just like "meh whatever" :) but I still tweaked my implementation a bit so it can never be bool anymore; as it's still better not to check property of boolean anyway.